### ISSUE TITLE
midori.profile: ad-blocker not working properly

### ISSUE THREAD

**Issue by Rosika2** [netblue30/firejail#3129]
Hi altogether,

I´ve encountered the following problem: When running the midori web-browser with `firejail midori ` 
it loads the _midori.profile_ which is correct. 

Midori runs alright except the issue with the _ad-blocker 2.0(0.4.1)_, which is enabled in midori by default.
The ad-blocker is active but doesn´t seem to function.

Example: when running  `firejail midori`  an then loading 
`https://www.linuxquestions.org/questions/showthread.php?p=6072547#post6072547`

the respective page is displayed **with** ads and the ad-blocker shield is **green**. So the ad-blocker is active but doesn´t seem to work.

When displaying the same page running  `firejail --noprofile midori `
the page is displayed **without** ads and the ad-blocker shield is **yellow**. That´s the way it should be.
So I guess some setting within the _midori.profile_ prevents the ad-blocker from kicking in accurately.

What might be the issue there?

Thanks a lot in advance.
Greetings.
Rosika

P.S.:
- midori & firejail running on BodhiLinux in virtual machine
- host: Lubuntu 18.04.3 LTS, 64 bit

**Comment by rusty-snake** [comment]
> What might be the issue there?

Probably the ad-blocker works, but has no rules.

We can separate the profile in there parts that can cause this:
1. the whitelist
2. the blacklist (`include disable-*.inc`)
3. the rest (`caps.drop all` to `disable-mnt`)

**Comment by Rosika2** [comment]
@rusty-snake:
Thank you so much for your help again.

I tried everything you suggested and it turned out to be `include disable-programs.inc`. 
That´s the only line I had to comment out.
So `# include disable-programs.inc` enables the pop-up blocker to read the rules. Its icon is yellow and ads aren´t shown.

Thanks again for your really fast quality help. It´s much appreciated.

Greetings.
Rosika

**Comment by rusty-snake** [comment]
@Rosika2 in general it is a bad idea to comment `include disable-progams.inc`. Here is it not so bad because midori is a whitelisting profile. disable-progams blacklists a lot of program path in $HOME, you could enable it and add a noblacklist command if you know the path. Use `journalctl --grep="blacklist violation" --reverse` to see logged access tries to blacklisted paths (from sandboxes with `tracelog`).

**Comment by Rosika2** [comment]
@rusty-snake:
Hello again and thanks a lot for the additional info.

I looked up my **journalctl**-entries. What worked for me was the command `journalctl | grep "blacklist violation"` which gave me as a result
`Jan 07 15:28:56 rosika2-Standard-PC-i440FX-PIIX-1996 firejail[2035]: blacklist violation - sandbox 2024, exe midori, syscall access, path /home/rosika2/.cache/midori`
Thus I got the respective path.
Afterwards I enabled `include disable-progams.inc` again and added  `noblacklist /home/rosika2/.cache/midori`.
And it works as desired. The ad-blocker reads the rules and applies them accordingly (yellow symbol) .

Thanks for explaining the inner workings of the profile.

Greetings.
Rosika
